cli.js

```diff
@@ -4,112 +4,224 @@ var fs = require('fs');
+		'  imagemin images build',
+		'  imagemin foo.png > foo-optimized.png',
+		'  imagemin < foo.png > foo-optimized.png',
+		'  imagemin -P pngquant foo.png > foo-optimized.png',
 		'',
 		'Options',
-		'  -i, --interlaced                    Interlace gif for progressive rendering',
-		'  -o, --optimizationLevel <number>    Optimization level between 0 and 7',
-		'  -p, --progressive                   Lossless conversion to progressive'
+		'  -p, --plugin                        override the default plugins',
+		'  -o, --output                        specify the output directory',
+		'      --interlaced                    interlace gif for progressive rendering',
+		'      --optimization-level <number>   optimization level between 0 and 7',
+		'      --progressive                   lossless conversion to progressive',
+		'  -v, --verbose                       logs each file as it gets minified',
+		'  -h, --help                          show this help'
 	]
 }, {
 	boolean: [
 		'interlaced',
-		'progressive'
+		'progressive',
+		'verbose'
 	],
 	string: [
-		'optimizationLevel'
+		'plugin',
+		'optimizationLevel',
+		'install'
 	],
 	alias: {
-		i: 'interlaced',
-		o: 'optimizationLevel',
-		p: 'progressive'
+		p: 'plugin',
+		o: 'output',
+		v: 'verbose'
 	}
 });
 
-function isFile(path) {
-	if (/^[^\s]+\.\w*$/.test(path)) {
-		return true;
-	}
+function exitWithMessage(message) {
+	console.error(Array.isArray(message) ? message.join('\n') : message);
+	process.exit(1);
+}
+
+var DEFAULT_PLUGINS = ['gifsicle', 'jpegtran', 'optipng', 'svgo'];
+var cwd = process.cwd();
 
-	try {
-		return fs.statSync(path).isFile();
-	} catch (err) {
-		return false;
+function run(src, dest, destName) {
+	var plugins = cli.flags.plugin || DEFAULT_PLUGINS;
+	if (!Array.isArray(plugins)) {
+		plugins = [plugins];
 	}
-}
 
-function run(src, dest) {
-	var imagemin = new Imagemin()
-		.src(src)
-		.use(Imagemin.gifsicle(cli.flags))
-		.use(Imagemin.jpegtran(cli.flags))
-		.use(Imagemin.optipng(cli.flags))
-		.use(Imagemin.svgo());
+	var imagemin = new Imagemin().src(src);
+
+	plugins.forEach(function (name) {
+		var plugin;
+		switch (name) {
+			case 'gifsicle':
+			case 'jpegtran':
+			case 'optipng':
+				plugin = Imagemin[name](cli.flags);
+				break;
+			case 'svgo':
+				plugin = Imagemin.svgo();
+				break;
+			default:
+				// workaround pngquant bug
+				var opts = {};
+				for (var key in cli.flags) {
+					if (name !== 'pngquant' && key !== 'verbose') {
+						opts[key] = cli.flags[key];
+					}
+				}
 
-	if (process.stdout.isTTY) {
-		imagemin.dest(dest ? dest : 'build');
+				try {
+					plugin = require('imagemin-' + name)(opts);
+				} catch (err) {
+					exitWithMessage([
+						'Unknown plugin ' + name,
+						'',
+						'Maybe you forgot to install the plugin?',
+						'You can install it with:',
+						'  npm install -g imagemin-' + name
+					]);
+				}
+				break;
+		}
+
+		imagemin.use(plugin);
+	});
+
+	var isBuffer = src instanceof Buffer;
```

Comment:
And I think there is no need to assign it to a temporary variable. https://github.com/imagemin/imagemin-cli/pull/3/files#r39933387